internal/provider/azure/azure.go

```diff
@@ -123,3 +134,365 @@ func (p *AzureProvider) getManagedZone(ctx context.Context, zoneID string) (prov
 func (p *AzureProvider) createManagedZone(_ context.Context, _ *v1alpha1.ManagedZone) (provider.ManagedZoneOutput, error) {
 	return provider.ManagedZoneOutput{}, nil
 }
+
+// Records gets the current records.
+//
+// Returns the current records or an error if the operation failed.
+func (p *AzureProvider) Records(ctx context.Context) (endpoints []*externaldnsendpoint.Endpoint, _ error) {
+	zones, err := p.Zones(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	p.logger.Info("getting records from azure")
```

Comment:
prob needs to be a bit more informative if we want to keep it

Reply:
this is very much debugging haha, there is probably a lot of log statements I need to clean up yet.